inst/private/doctest_collect.m

```diff
@@ -135,6 +141,8 @@
   targets = [target];
 elseif strcmp(type, 'class')
   targets = collect_targets_class(what, depth);
+elseif strcmp (type, 'octfile')
+  targets = collect_targets_octfile (what, depth);
```

Comment:
This is way nicer than my recursive approach. I didn't read enough of the code to see how to build up the targets list, I just saw the way directories were handled and followed that logic. I like this a lot.